api/v1alpha1/artifact_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2024 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	v1 "github.com/fluxcd/source-controller/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// BucketKind is the string representation of a Bucket.
+	ArtifactKind = "Artifact"
+)
+
+// ArtifactSpec defines the desired state of Artifact
+type ArtifactSpec struct {
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	CurrentVersion string `json:"currentVersion,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Versions map[string]*v1.Artifact `json:"versions,omitempty"`
+}
```

Comment:
```suggestion
// ArtifactSpec defines the desired state of Artifact
type ArtifactSpec v1.Artifact
```

Reply:
> status is not made to be backup / stateful. So customers can't rely on it.

For sure, no customer should relay on anything from this custom resource, status or not is irelevant. This custom resource is a side-efect of some 3rd-party source-controller that should fully manage the artifact. This type of object is not "desired state" it shouldn't be included in the backup. You should backup what ever custom resource your controller uses for "desired state".

Reply:
It's ok for me, just wanted to know if the desired state should be managed by source controller and just an information storage :)

And yes, I'm developing a source-controller for Omaha / [Nebraska](https://github.com/flatcar/nebraska), so I will need a way to interact with consumers controllers :)

Reply:
Flux source-controller plays no role here, this `ArtifactX` type is to enable other source controllers to feed data to Flux kustomize-controller.